### ISSUE TITLE
Failed to run 00_getting_started.py

### ISSUE THREAD

**Issue by divyaprakash0426** [radical-cybertools/radical.pilot#1737]
I tried to run the getting started file multiple times but I get errors. My output looks something like this:
```
================================================================================
 Getting Started (RP version 0.50.8)                                            
================================================================================

new session: [rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0005]     \
database   : [mongodb://ks_rp:ks_rp123@ds119853.mlab.com:19853/ksiddhant_rp]  ok
read config                                                                   ok

--------------------------------------------------------------------------------
submit pilots                                                                   

create pilot manager                                                          ok
submit 1 pilot(s)
        [local.localhost:32]
                                                                              ok

--------------------------------------------------------------------------------
submit units                                                                    

create unit manager                                                           ok
add 1 pilot(s)                                                                ok
create 32 unit description(s)
        ................................                                      ok
submit 32 unit(s)
        ................................                                      ok

--------------------------------------------------------------------------------
gather results                                                                  

wait for 32 unit(s)
        --------------------------------                                      ok

--------------------------------------------------------------------------------
finalize                                                                        

closing session rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0005    \
close unit manager                                                            ok
close pilot manager                                                            \
wait for 1 pilot(s)
                                                                         timeout
                                                                              ok
+ rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0005 (json)
- pilot.0000 (profiles)
- pilot.0000 (logfiles)
session lifetime: 24.4s                                                       ok

--------------------------------------------------------------------------------
```
My radical stack is:
```
> python               : 2.7.15rc1
  pythonpath           : 
  virtualenv           : /home/divyaprakash/RADICALPilot
  radical.pilot        : 0.50.8
  radical.utils        : 0.50.2
  saga                 : 0.50.0
```

A look into the error logs shows this:

```
pmgr.0000.log:2018-10-08 09:48:00,764: pmgr.0000           : MainProcess                     : pmgr.0000.subscriber._state_sub_cb: ERROR   : [Callback]: pilot 'pilot.0000' failed (exit on error)
pmgr.0000.log:2018-10-08 09:48:00,764: pmgr.0000           : MainProcess                     : pmgr.0000.subscriber._state_sub_cb: ERROR   : abort: RuntimeError('pilot pilot.0000 failed - fatal!',)
rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0001.log:2018-10-08 09:48:12,246: rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0001: MainProcess                     : MainThread     : ERROR   : failed to fet profile for pilot.0000
```

The description of the errors are as follows:

1.  **pmgr.0000.log:**

```
2018-10-08 09:48:00,764: pmgr.0000           : MainProcess                     : pmgr.0000.subscriber._state_sub_cb: ERROR   : [Callback]: pilot 'pilot.0000' failed (exit on error)

2018-10-08 09:48:00,764: pmgr.0000           : MainProcess                     : pmgr.0000.subscriber._state_sub_cb: ERROR   : abort: RuntimeError('pilot pilot.0000 failed - fatal!',)

Traceback (most recent call last):

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/utils/threads.py", line 375, in _run

    if not self.work_cb():

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/utils/component.py", line 1124, in work_cb

    ret = self._cb(topic=topic, msg=m)

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/pilot_manager.py", line 296, in _state_sub_cb

    if not self._update_pilot(thing, publish=False):

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/pilot_manager.py", line 337, in _update_pilot

    self._pilots[pid]._update(pilot_dict)

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/compute_pilot.py", line 185, in _update

    else      : cb(self, self.state)

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/compute_pilot.py", line 133, in _default_state_cb

    raise RuntimeError('pilot %s failed - fatal!' % self.uid)

RuntimeError: pilot pilot.0000 failed - fatal!
```

2. **rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0001.log**

```
2018-10-08 09:48:12,246: rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0001: MainProcess                     : MainThread     : ERROR   : failed to fet profile for pilot.0000

Traceback (most recent call last):

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/utils/session.py", line 168, in fetch_profiles

    profiles = sandbox.list('*.prof')

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/namespace/directory.py", line 243, in list

    return self._adaptor.list (pattern, flags, ttype=ttype)

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/adaptors/cpi/decorators.py", line 57, in wrap_function

    return sync_function (self, *args, **kwargs)

  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/adaptors/shell/shell_file.py", line 473, in list

    % (ret, out))

NoSuccess: failed to list(): (2)(/bin/ls: cannot access '*.prof': No such file or directory

) (/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/adaptors/shell/shell_file.py +473 (list)  :  % (ret, out)))
```

- **My code for 00_getting_started.py is :**

```
#!/usr/bin/env python



__copyright__ = 'Copyright 2013-2014, http://radical.rutgers.edu'

__license__   = 'MIT'



import os

import sys

import time



import radical.pilot as rp

import radical.utils as ru



dh = ru.DebugHelper()



# ------------------------------------------------------------------------------

#

# READ the RADICAL-Pilot documentation: http://radicalpilot.readthedocs.org/

#

# ------------------------------------------------------------------------------





# ------------------------------------------------------------------------------

#

if __name__ == '__main__':



    # we use a reporter class for nicer output

    report = ru.Reporter(name='radical.pilot')

    report.title('Getting Started (RP version %s)' % rp.version)



    # use the resource specified as argument, fall back to localhost

    if   len(sys.argv)  > 2: report.exit('Usage:\t%s [resource]\n\n' % sys.argv[0])

    elif len(sys.argv) == 2: resource = sys.argv[1]

    else                   : resource = 'local.localhost'



    # Create a new session. No need to try/except this: if session creation

    # fails, there is not much we can do anyways...

    session = rp.Session()



    # all other pilot code is now tried/excepted.  If an exception is caught, we

    # can rely on the session object to exist and be valid, and we can thus tear

    # the whole RP stack down via a 'session.close()' call in the 'finally'

    # clause...

    try:



        # read the config used for resource details

        report.info('read config')

        config = ru.read_json('%s/config.json' % os.path.dirname(os.path.abspath(__file__)))

        report.ok('>>ok\n')



        report.header('submit pilots')



        # Add a Pilot Manager. Pilot managers manage one or more ComputePilots.

        pmgr = rp.PilotManager(session=session)



        # Define an [n]-core local pilot that runs for [x] minutes

        # Here we use a dict to initialize the description object

        pd_init = {'resource'      : resource,

                   'runtime'       : 15,  # pilot runtime (min)

                   'exit_on_error' : True,

                   'project'       : config[resource]['project'],

                   'queue'         : config[resource]['queue'],

                   'access_schema' : config[resource]['schema'],

                   'cores'         : config[resource]['cores']

                  }

        pdesc = rp.ComputePilotDescription(pd_init)



        # Launch the pilot.

        pilot = pmgr.submit_pilots(pdesc)



        report.header('submit units')



        # Register the ComputePilot in a UnitManager object.

        umgr = rp.UnitManager(session=session)

        umgr.add_pilots(pilot)



        # Create a workload of ComputeUnits.

        # Each compute unit runs '/bin/date'.



        n = 32  # number of units to run

        report.info('create %d unit description(s)\n\t' % n)



        cuds = list()

        for i in range(0, n):



            # create a new CU description, and fill it.

            # Here we don't use dict initialization.

            cud = rp.ComputeUnitDescription()

            cud.executable       = '/bin/date'

            cud.gpu_processes    = 0

            cud.cpu_processes    = 2

            cud.cpu_threads      = 2

            cud.cpu_process_type = rp.OpenMP

            cud.cpu_thread_type  = rp.OpenMP

            cuds.append(cud)

            report.progress()

        report.ok('>>ok\n')



        # Submit the previously created ComputeUnit descriptions to the

        # PilotManager. This will trigger the selected scheduler to start

        # assigning ComputeUnits to the ComputePilots.

        umgr.submit_units(cuds)



        # Wait for all compute units to reach a final state (DONE, CANCELED or FAILED).

        report.header('gather results')

        umgr.wait_units()





    except Exception as e:

        # Something unexpected happened in the pilot code above

        report.error('caught Exception: %s\n' % e)

        ru.print_exception_trace()

        raise



    except (KeyboardInterrupt, SystemExit) as e:

        # the callback called sys.exit(), and we can here catch the

        # corresponding KeyboardInterrupt exception for shutdown.  We also catch

        # SystemExit (which gets raised if the main threads exits for some other

        # reason).

        ru.print_exception_trace()

        report.warn('exit requested\n')



    finally:

        # always clean up the session, no matter if we caught an exception or

        # not.  This will kill all remaining pilots.

        report.header('finalize')

        session.close(download=True)



    report.header()

# ------------------------------------------------------------------------------
```

**Comment by andre-merzky** [comment]
Thanks for the detailed report!  I am afraid that our default MongoDB (which is used by default  unless a different DBURL is set) has a problem: it is full, but we are unable to clean it out.  I opened a ticket with mlab.com (which hosts the DB), and will update you as soon as I hear back.  Sorry for the troubles!

**Comment by andre-merzky** [comment]
The DB has been cleaned out now - can you please give it another try, and report back?  Thanks!

**Comment by divyaprakash0426** [comment]
I already used a different MongoDB URL : RADICAL_PILOT_DBURL=mongodb://ks_rp:ks_rp123@ds119853.mlab.com:19853/ksiddhant_rp

I gave it another try but I got the same errors

**Comment by andre-merzky** [comment]
Ah, bugger - that would have been too easy ;-)

So, the messages you see indicate that the pilot failed.  Please have a look into `$HOME/radical.pilot.sandbox`.  You should find a directory there with the same name as your session (`rp.session.divyaprakash-VirtualBox.divyaprakash.017812.0001` in the case above).  There you'll find a subdirectory names `pilot.0000`.  That is the pilot sandbox, i.e. the area where the pilot stores its logfiles, executes the compute units, etc.  Please create a tarball from this pilot sandbox, and attach it to this ticket.

Many thanks, Andre.

**Comment by divyaprakash0426** [comment]
Hello Andre, I created the zip file. I wasn't able to upload the .tar file

[pilot.0000.zip](https://github.com/radical-cybertools/radical.pilot/files/2470004/pilot.0000.zip)

**Comment by divyaprakash0426** [comment]
Wow!! Thanks a lot. It worked. However, I still get one last error which I got the last time as well along with two other errors which are now resolved after installing curl.

`rp.session.divyaprakash-VirtualBox.divyaprakash.017815.0008.log:2018-10-11 19:04:28,168: rp.session.divyaprakash-VirtualBox.divyaprakash.017815.0008: MainProcess                     : MainThread     : ERROR   : failed to fet profile for pilot.0000`

The - pilot.0000 (profiles) is the only place where I couldn't get the OK. Rest everything is fine.

The description of the error is as follows:
`2018-10-11 19:04:28,168: rp.session.divyaprakash-VirtualBox.divyaprakash.017815.0008: MainProcess                     : MainThread     : ERROR   : failed to fet profile for pilot.0000
Traceback (most recent call last):
  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/radical/pilot/utils/session.py", line 168, in fetch_profiles
    profiles = sandbox.list('*.prof')
  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/namespace/directory.py", line 243, in list
    return self._adaptor.list (pattern, flags, ttype=ttype)
  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/adaptors/cpi/decorators.py", line 57, in wrap_function
    return sync_function (self, *args, **kwargs)
  File "/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/adaptors/shell/shell_file.py", line 473, in list
    % (ret, out))
NoSuccess: failed to list(): (2)(/bin/ls: cannot access '*.prof': No such file or directory
) (/home/divyaprakash/RADICALPilot/local/lib/python2.7/site-packages/saga/adaptors/shell/shell_file.py +473 (list)  :  % (ret, out)))`

**Comment by andre-merzky** [comment]
Please set the environment variable `RADICAL_PROFILE=TRUE` to get profiles.  But also, please leave that ticket open - the examples should not fail if that env variable is not set, this needs fixing.  Thanks!

**Comment by divyaprakash0426** [comment]
Everything works perfect now!!  Thank you.

**Comment by lee212** [comment]
Summary (after reading the thread) of this issue, 1) default Mongo was full (cleaned now), 2) curl was missing on a client machine (installed), 3) `RADICAL_PROFILE` env variable has to be set.

@andre-merzky, is this going to be resolved in v0.60?

**Comment by andre-merzky** [comment]
I think this needs no further action, and can be resolved.  Just a  note though that I repeatedly did not manager to clean out the mlab DB over the last months - that service seems to become less reliable.  We may need an action here at some point...
Thanks!
